--- Fiona-1.9.4.orig/tests/test_memoryfile.py	2023-09-25 09:39:39.573168915 -0500
+++ Fiona-1.9.4/tests/test_memoryfile.py	2023-09-25 09:40:19.754995996 -0500
@@ -376,11 +376,11 @@
 def test_listdir_zipmemoryfile(bytes_coutwildrnp_zip):
     """Test list directories of a zipped memory file."""
     with ZipMemoryFile(bytes_coutwildrnp_zip) as memfile:
-        assert memfile.listdir() == [
-            "coutwildrnp.shp",
-            "coutwildrnp.shx",
+        assert sorted(memfile.listdir()) == [
             "coutwildrnp.dbf",
             "coutwildrnp.prj",
+            "coutwildrnp.shp",
+            "coutwildrnp.shx",
         ]
 
 
