--- fiona-1.10.1.orig/fiona/fio/cat.py	2025-12-04 10:13:34.764012425 -0600
+++ fiona-1.10.1/fiona/fio/cat.py	2025-12-29 16:32:23.814944881 -0600
@@ -55,8 +55,6 @@
     default=False,
     help="Optionally cut geometries at the anti-meridian. To be used only for a geographic destination CRS.",
 )
-@click.option('--where', default=None,
-              help="attribute filter using SQL where clause")
 @options.open_opt
 @click.pass_context
 @with_context_env
