--- fiona-1.10.1.orig/fiona/_vendor/snuggs.py	2025-12-04 10:13:34.763012393 -0600
+++ fiona-1.10.1/fiona/_vendor/snuggs.py	2025-12-29 17:03:57.731048567 -0600
@@ -34,7 +34,7 @@
 
 from pyparsing import (  # type: ignore
     Keyword,
-    oneOf,
+    one_of,
     Literal,
     QuotedString,
     ParseException,
@@ -158,7 +158,7 @@
 string = QuotedString("'") | QuotedString('"')
 lparen = Literal("(").suppress()
 rparen = Literal(")").suppress()
-op = oneOf(" ".join(op_map.keys())).set_parse_action(
+op = one_of(" ".join(op_map.keys())).set_parse_action(
     lambda source, loc, toks: op_map[toks[0]]
 )
 
@@ -177,7 +177,7 @@
 # variables.
 func = Regex(r"(?<=\()[{}]+".format(alphanums + "_")).set_parse_action(resolve_func)
 
-higher_func = oneOf(" ".join(higher_func_map.keys())).set_parse_action(
+higher_func = one_of(" ".join(higher_func_map.keys())).set_parse_action(
     lambda source, loc, toks: higher_func_map[toks[0]]
 )
 
