--- Fiona-1.8.12.orig/setup.py	2019-12-05 03:08:55.000000000 -0300
+++ Fiona-1.8.12/setup.py	2019-12-05 08:50:47.017441535 -0300
@@ -280,7 +280,7 @@
     'click-plugins>=1.0',
     'six>=1.7',
     'munch',
-    'argparse; python_version <= "3.4"',
+    'argparse; python_version <= "2.6"',
-     'ordereddict; python_version <= "2.7"',
+     'ordereddict; python_version <= "2.6"',
     'enum34; python_version < "3.4"'
 ]
