--- Fiona-1.8.15.orig/fiona/_loading.py	2020-07-17 13:52:48.000000000 -0300
+++ Fiona-1.8.15/fiona/_loading.py	2020-09-05 09:56:06.039217756 -0300
@@ -13,45 +13,14 @@
 # os.add_dll_directory.
 # see https://github.com/Toblerity/Fiona/issues/851
 
-dll_directories = []
-
-
-def directory_contains_gdal_dll(path):
-    """ Checks if a directory contains a gdal dll """
-    return len(glob.glob(os.path.join(path, "gdal*.dll"))) > 0
-
-
-def search_gdal_dll_directories():
-    """ Search for gdal dlls
-
-        Finds directories in PATH containing gdal.dll.
-    """
-
-    # Parse PATH for gdal/bin
-    for path in os.getenv('PATH', '').split(os.pathsep):
-
-        if directory_contains_gdal_dll(path):
-            dll_directories.append(path)
-
-    if len(dll_directories) == 0:
-        log.warning("No dll directories found.")
-
-
-if platform.system() == 'Windows' and sys.version_info >= (3, 8):
-
-    # if loading of extension modules fails, search for gdal dll directories
-    try:
-        import fiona.ogrext
-    except ImportError as e:
-        search_gdal_dll_directories()
-
 
 @contextlib.contextmanager
 def add_gdal_dll_directories():
-
     dll_dirs = []
-    for dll_directory in dll_directories:
-        dll_dirs.append(os.add_dll_directory(dll_directory))
+    if platform.system() == 'Windows' and sys.version_info >= (3, 8):
+        dll_directory = os.path.join(os.path.dirname(__file__), '.libs')
+        if os.path.exists(dll_directory):
+            dll_dirs.append(os.add_dll_directory(dll_directory))
     try:
         yield None
     finally:
